src/components/Extensions/index.tsx

```diff
@@ -46,39 +49,16 @@ export interface Backend {
   extensionSpec?: ExtensionSpec;
 }
 
-function useFunctionsEmulator() {
-  const config = useEmulatorConfig('extensions');
-  return `http://${config.hostAndPort}`;
-}
-
-function useExtensionBackends(): ExtensionBackend[] {
-  const functionsEmulator = useFunctionsEmulator();
-
-  const fetcher = async () => {
-    const response = await fetch(`${functionsEmulator}/backends`);
-    const json = await response.json();
-    return json.backends;
-  };
-
-  const backends = useSwr(`list_backends`, fetcher, { suspense: true }).data;
-  return backends.filter(isExtensionBackend);
-}
-
-function isExtensionBackend(backend: Backend): backend is ExtensionBackend {
-  return (
-    !!backend.extensionInstanceId &&
-    ((!!backend.extension && !!backend.extensionVersion) ||
-      !!backend.extensionSpec)
-  );
-}
-
 const HydrateExtensions: React.FC = ({ children }) => {
   const backends = useExtensionBackends();
+  const specs = useExtensionRowSpecs();
 
   return (
-    <ExtensionsProvider extensionBackends={backends}>
-      {children}
-    </ExtensionsProvider>
+    <ExtensionsSpecProvider extensionRowSpecs={specs}>
+      <ExtensionsProvider extensionBackends={backends}>
```

Comment:
I'm a little confused here. I'd imagine a list of extension-backends at this level, but then at the list/details-level there would be hooks to get the scoped Spec